src/app/Table.elm

```diff
@@ -1,29 +1,91 @@
-module Table exposing (view, init, Msg)
+module Table exposing (view, init, initWithPlayersAndCup)
+
+{-|
+
+   This module handles the table of players,
+   and everything relating to that.
+
+-}
 
 import Html exposing (..)
+import Html.Events exposing (..)
+import Types exposing (PlayerID)
+import TableMsg exposing (Msg(..))
+import Dice
 
 
-{-
+type alias PlayerName =
+    String
 
-   This module handles the view rendering for the table of players
-   and everything relating to that.
 
--}
+type alias DiceCount =
+    Int
+
 
+type alias TablePlayer =
+    { id : PlayerID
+    , name : PlayerName
+    , diceCount : DiceCount
+    }
 
-type Model
-    = Model
 
+type alias TablePlayers =
+    List TablePlayer
 
-type Msg
-    = Msg
+
+type alias Model =
+    { tablePlayers : TablePlayers
+    , myCup : Dice.Cup
+    }
 
 
 init : Model
 init =
-    Model
+    { tablePlayers = noPlayers
+    , myCup = Dice.emptyCup
+    }
+
+
+initWithPlayersAndCup : List ( Int, String, Int ) -> List Int -> Model
+initWithPlayersAndCup tuples cup =
+    let
+        tupleToPlayer ( id, name, diceCount ) =
+            { id = id, name = name, diceCount = diceCount }
+    in
+        { tablePlayers = List.map tupleToPlayer tuples
+        , myCup = Dice.cupFromIntList cup
+        }
+
+
+noPlayers : TablePlayers
+noPlayers =
+    []
+
+
+testPlayers : TablePlayers
+testPlayers =
+    [ TablePlayer 1 "Joe" 6, TablePlayer 2 "Garrett" 5 ]
+
+
+view : (Msg -> msg) -> Model -> Html msg
```

Comment:
The top level `Msg` **has** to grow larger as we add more messages to the app. It's impossible not to, isn't it? The point is this *minimizes* the amount of changes. All local messages are contained within that one (line 57) sub-Msg:

```
    | TableAction TableMsg.Msg
```

So if anything it *minimizes* the amount of messages added to that `Msg` type.

I'm all ears if you think there's a better way to do this.

The Table is not *going to be* a page with 2 - 3 buttons on it, it'll have all of the application's table buttons on it, won't it?

And, it may be that we don't actually need a `TableAction` later on, in which case it will go away. I'm still experimenting with how this will work out, and seeing what we need for building the table.

This depends if we need mounted state, as well. 

However, my major concern here is that the pieces of code I'm changing are isolated mostly to the `Table` module (and its related) file(s).

Did you notice that? I've changed almost nothing in other files. That means more than one of us can be working on this at once.

--- 

This is *not* the translator pattern, by the way. I'm not building a custom dictionary of type mappings or anything like that. All I've done is more the `Html.map` *into* the component itself, and generisized that to `msg` rather than `Msg` which means things can be composed in multiple hierarchies as well as kept on a flat level.

I'm interested in doing this because if we can show to the rest of the Blake team that these types of modules and pieces of code in Elm can be more easily *composed*, *reused* and that we can easily work on things with *multiple* devs at once, it might be a contender for being adopted by more things, which would be more than nice. :)

I think maybe you're underestimating beginners.